src/vcpkg/versions.cpp

```diff
@@ -50,93 +51,185 @@ namespace vcpkg::Versions
         return hash<string>()(key.port_name) ^ (hash<string>()(key.version.to_string()) >> 1);
     }
 
-    ExpectedS<RelaxedVersion> RelaxedVersion::from_string(const std::string& str)
+    // 0|[1-9][0-9]*
+    static const char* parse_skip_number(const char* const s, uint64_t* const n)
     {
-        std::regex relaxed_scheme_match("^(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*))*");
-
-        if (!std::regex_match(str, relaxed_scheme_match))
+        const char ch = *s;
+        if (ch == '0')
         {
-            return Strings::format(
-                "Error: String `%s` must only contain dot-separated numeric values without leading zeroes.", str);
+            *n = 0;
+            return s + 1;
         }
-
-        return RelaxedVersion{str, Util::fmap(Strings::split(str, '.'), [](auto&& strval) -> uint64_t {
-                                  return as_numeric(strval).value_or_exit(VCPKG_LINE_INFO);
-                              })};
+        if (ch < '1' || ch > '9')
+        {
+            return nullptr;
+        }
+        size_t i = 1;
+        for (; Parse::ParserBase::is_ascii_digit(s[i]); ++i)
+        {
+        }
+        *n = as_numeric({s, i}).value_or_exit(VCPKG_LINE_INFO);
+        return s + i;
     }
 
-    ExpectedS<SemanticVersion> SemanticVersion::from_string(const std::string& str)
+    ExpectedS<RelaxedVersion> RelaxedVersion::from_string(const std::string& str)
     {
-        // Suggested regex by semver.org
-        std::regex semver_scheme_match("^(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)"
-                                       "(?:-((?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9][0-9]*|[0-9]"
-                                       "*[a-zA-Z-][0-9a-zA-Z-]*))*))?"
-                                       "(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$");
+        // (0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))*;
```

Comment:
Is the regex supposed to end with a semicolon here?

Reply:
Probably not, the parsing changes are cherry-picked from Robert's PR to speed-up search.
I think the `;` is a left-over from commenting a call to `std::regex_match()`.

Reply:
Yep, that's my bad.